test/payGasExtension.js

```diff
@@ -0,0 +1,115 @@
+const utils = require('./utils')
+const solc = require('solc')
+
+const GnosisSafe = artifacts.require("./GnosisSafePersonalEdition.sol");
+const MultiSend = artifacts.require("./libraries/MultiSend.sol")
+const Battery = artifacts.require("./libraries/Battery.sol")
+const CreateAndAddExtension = artifacts.require("./libraries/CreateAndAddExtension.sol");
+const ProxyFactory = artifacts.require("./ProxyFactory.sol");
+const PayGasExtension = artifacts.require("./extensions/PayGasExtension.sol");
+const TransactionWrapper = web3.eth.contract([{"constant":false,"inputs":[{"name":"to","type":"address"},{"name":"value","type":"uint256"},{"name":"data","type":"bytes"}],"name":"send","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"}]);
+
+contract('PayGasExtension', function(accounts) {
+
+    let gnosisSafe
+    let extension
+    let multiSend
+    let battery
+    let lw
+    let tw = TransactionWrapper.at(1)
+
+    const CALL = 0
+
+    let payAndExecuteTransaction = async function(target, value, useBattery) {
+      let nonce = await gnosisSafe.nonce()
+      let transactionHash = await gnosisSafe.getTransactionHash.call(target, value, 0, CALL, nonce)
+      // Confirm transaction with signed messages
+      let sigs = utils.signTransaction(lw, [lw.accounts[0], lw.accounts[2]], transactionHash)
+      // Generate price signature
+      let estimate = await gnosisSafe.executeTransaction.estimateGas(target, value, 0, CALL, sigs.sigV, sigs.sigR, sigs.sigS) + 30000
```

Comment:
we add 30k to include the costs for the payment itself. We cannot estimate the final transaction as we don't have a valid signature for that